src/main/java/com/mykaarma/urlshortener/scheduled/HashGenerationJob.java

```diff
@@ -0,0 +1,79 @@
+package com.mykaarma.urlshortener.scheduled;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import com.mykaarma.urlshortener.exception.ShortUrlException;
+import com.mykaarma.urlshortener.persistence.AvailableHashPoolAdapter;
+import com.mykaarma.urlshortener.persistence.HashArchiveAdapter;
+import com.mykaarma.urlshortener.util.UrlServiceUtil;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+@EnableAsync
+public class HashGenerationJob {
+
+	@Value("${hash_count_threshold:100000}")
+	private int hashCountThreshold;
+	
+	@Value("${hash_generation_rate:100000}")
+	private int hashGenerationRate;
+	
+	@Value("${hash_length:8}")
+	private int hashLength;
+	
+	private AvailableHashPoolAdapter availableHashPoolAdapter;
+	private HashArchiveAdapter hashArchiveAdapter;
+	private UrlServiceUtil urlServiceUtil;
+	
+	@Autowired
+	public HashGenerationJob(AvailableHashPoolAdapter availableHashPoolAdapter, HashArchiveAdapter hashArchiveAdapter, UrlServiceUtil urlServiceUtil) {
+		this.availableHashPoolAdapter = availableHashPoolAdapter;
+		this.hashArchiveAdapter = hashArchiveAdapter;
+		this.urlServiceUtil = urlServiceUtil;
+	}
+	
+	@Async
+	@Scheduled(cron = "${hash_generation_cron}")
+	public void runHashesGenerationJob() throws ShortUrlException {
```

Comment:
@AadiS11 What happens with our Jobs when our URL shortener is running on multiple pods?

Reply:
Added lock